### PR TITLE
Release builds for samples

### PR DESCRIPTION
**Pull request by yschimke**
Fixes https://github.com/android/wear-os-samples/issues/238#issuecomment-1065984919

### CODE REVIEW

**Comment by LouisCAD** [comment]
Could you also add an `isDefault = true` to the release buildTypes of the Compose app so it's the default selected by Android Studio?

**Comment by LouisCAD** [comment]
I'm also thinking that it might not be enough: When I tried the samples, I built them by running `.gradlew installDebug`.

I wonder if R8 shouldn't be enabled on debug builds too by default, albeit with obfuscation disabled.

Thinking a bit deeper, I guess it'd make sense to have a "Slow mode" ribbon, like debug Flutter apps have when `minifyEnabled` is left/set to false. That'd make people search for the non slow mode that requires minification to be turned on.

File: .github/workflows/AlwaysOnKotlin.yml [hunk 1]
```diff
@@ -119,3 +119,36 @@ jobs:
...
+      - name: Generate cache key
+        run: ./scripts/checksum.sh $SAMPLE_PATH checksum.txt
+
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches/modules-*
+            ~/.gradle/caches/jars-*
+            ~/.gradle/caches/build-cache-*
+          key: gradle-${{ hashFiles('checksum.txt') }}
```
Comment (Goooler) [inline .github/workflows/AlwaysOnKotlin.yml:144]: Use `gradle/gradle-build-action@v2` ?

**Review by shumelchyk — APPROVED** [review]
Yuri, PR looks good, let's also handle suggestions from @LouisCAD

**Comment by yschimke** [comment]
I think it's non controversial to change the default to release for any compose modules.  But I'll hold off on the Slow mode ribbon.

**Comment by yschimke** [comment]
I'm not confident about the behaviour of unobfuscated debug builds for general debugging, so I'll hold off on that for now.

**Comment by shumelchyk** [comment]
> so I'll hold off on that for now
I agree on this because Compose for WearOS is still in alpha version and we have planned improvements on the roadmap.

**Comment by yschimke** [comment]
This breaks the previews.  Might add a comment in the code.  We can't have both unless we ship the Preview classes in release.

https://github.com/android/wear-os-samples/pull/247#discussion_r832465742

**Pull request merged**